### ISSUE TITLE
Google Home Mini - Nora is not available

### ISSUE THREAD

**Issue by syoma755** [andrei-tatar/node-red-contrib-nora#37]
Hi guys

I was trying to use Nora in connection with Google Home Mini. Everything went well until one - two weeks ago when Google Home mini stopped reacting on voice commends. Instead it always responds „Nora is not available“. Restart doesn‘t help.
The funny story is that at the same time iOS Google Home App works just fine with NORA devices. Only speaker doesn’t. 

What could be the problem?

**Comment by CapnScabby** [comment]
I have a similar Issue. The google home app on android can control all my Nora Devices, but the google home mini's in my house frequently complain that Nora Can't be reached

I am based in South East Asia, perhaps it is a server latency issue?

**Comment by andrei-tatar** [comment]
> I am based in South East Asia, perhaps it is a server latency issue?

Could be. Does node-red manage to establish a websocket connection to NORA and keep it open?

**Comment by HashDot** [comment]
Today the same issue occurs me in Germany. NORA isn't reachable with my GHM and the GH App.

**Comment by andrei-tatar** [comment]
You need to update the token in Node-RED with a new one from https://node-red-google-home.herokuapp.com/

Also, you need to unlink NORA (in your Google Home app) and link it again.

Sorry for the inconvenience... Everyone will need to do that.

**Comment by HashDot** [comment]
Hej @andrei-tatar thank you for your fast answer. Works like a charm.

**Comment by syoma755** [comment]
> You need to update the token in Node-RED with a new one from
> https://node-red-google-home.herokuapp.com/
>
> Also, you need to unlink NORA (in your Google Home app) and link it again.
>
> Sorry for the inconvenience... Everyone will need to do that.
>
Hi Andrei, will it resolve the "Nora is not reachable issue"? What was the
problem?

**Comment by HashDot** [comment]
@syoma755 it solves the problem "Nora is not reachable"

**Comment by syoma755** [comment]
I have relinked Nora to Node-Red and revoked Token. Also tried to find new devices in Nora binding in Google Home app - app reported, that Nora service successfully linked. I can control Nora devices from the Google Home app without a problem.

But there is still some problem:
I have two Google Home Minis in two rooms. On one the Nora is working now, but from time to time GHM is reporting that "Nora is not available". If command is repeated, it is usually executed. 
But the other GHM continuously reports "Nora is not available" - I couldn't make it work so far. 

Can you suggest what can be wrong? How to make my GHMs work correctly with NORA?

**Comment by CapnScabby** [comment]
@syoma755 are the google home minis on the same wifi? Is one significantly farther from the router?

Google will give the not responding error if it doesn't receive a reply in a window, but the respone window is pretty narrow, so if anything is delaying the packets (in my case being on the other side of the planet from the nora server) can cause this error, but usually I find that even when google says the service is unavailable the command still gets executed.

**Comment by syoma755** [comment]
The minis are in the same Wi-Fi, but in different rooms. The one, which doesn't work, is a bit farther from router, than the other. 
The one thing, which might affect this, is that on non-working mini I have two accounts. But it's still in the same house and works fine with other appliances (I have also AC climate, connected via Smart Things to Google)

**Comment by CapnScabby** [comment]
Did you link nora to all the google accounts? That might be necessary. The other thing is that nora is giving out API keys associated with distinct accounts. Is your nora connector handling multiple IDs or just one?

**Comment by syoma755** [comment]
> Did you link nora to all the google accounts?
I have no Idea why I should do it and how to check it. Off course I have linked NORA only to my main Google Account and using this account I have generated a token for Node-Red. But what should I do with second account? How to link it to Nora using same token, if in Google Home App I logged in under main account?

**Comment by CapnScabby** [comment]
Yeah, that I dunno. Maybe if you used github to create your nora token you could use that same github account with both google home accounts?
